src/components/EmailResetForm.tsx

```diff
@@ -0,0 +1,152 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import Button from "./Button-Link/Button/Button";
+import { faX } from "@fortawesome/free-solid-svg-icons";
+import { useState } from "react";
+
+const EmailResetForm = ({
+  initialValue,
+  onSave,
+  label,
+}: {
+  initialValue: string;
+  onSave: (email: string, password: string) => void;
+  label: string;
+}) => {
+  const [passwordValue, setPasswordValue] = useState("");
+  const [emailValue, setEmailValue] = useState(initialValue);
+  const [isEditable, setIsEditable] = useState(false);
+  const [wizard, setWizard] = useState<
+    | {
+        type: "initial";
+      }
+    | {
+        type: "loading";
+      }
+    | {
+        type: "error";
+        message: string;
+      }
+  >({ type: "initial" });
```

Comment:
This is not ok. We have 2 states that should be merged into one. We have `wizard` and `isEditable`.

Right now we can be in `loading` and also have `isEditable: false`. But that doesn't make sense, right?